From 1bd5cc27db2777447a46ba6e44e318ceafd5821f Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:30:17 -0400
Subject: [net] openvswitch: Add skb_clone NULL check for the sampling action

Message-id: <a34c9068add22b770bc0318dd0b2dd117f03a4c9.1410531783.git.jbenc@redhat.com>
Patchwork-id: 94015
O-Subject: [RHEL7.1 net PATCH 44/50] openvswitch: Add skb_clone NULL check for the sampling action.
Bugzilla: 1110384
RH-Acked-by: Flavio Leitner <fbl@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110384
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

commit d9e0ecb81417c34ef8c02a6880d23c362300cda0
Author: Andy Zhou <azhou@nicira.com>
Date:   Thu Jul 17 15:17:54 2014 -0700

    openvswitch: Add skb_clone NULL check for the sampling action.

    Fix a bug where skb_clone() NULL check is missing in sample action
    implementation.

    Signed-off-by: Andy Zhou <azhou@nicira.com>
    Signed-off-by: Pravin B Shelar <pshelar@nicira.com>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/openvswitch/actions.c b/net/openvswitch/actions.c
index 5445d9d..df79994 100644
--- a/net/openvswitch/actions.c
+++ b/net/openvswitch/actions.c
@@ -482,6 +482,8 @@ static int sample(struct datapath *dp, struct sk_buff *skb,
   skb_get(skb);
  } else {
   sample_skb = skb_clone(skb, GFP_ATOMIC);
+  if (!sample_skb) /* Skip sample action when out of memory. */
+   return 0;
  }
 
  /* Note that do_execute_actions() never consumes skb.
-- 
1.7.1